changelog/0032-upgrade-to-elasticsearch-2.4.6.yml

```diff
@@ -1,102 +1,115 @@
+
+  0. For commands including `${ENV}` below to work, make sure to set `ENV` to the name of your environment
+  ```
+  ENV=${ENV}
+  ```
+
   1. Stop the site
   ```bash
-  commcare-cloud <env> downtime start
+  commcare-cloud ${ENV} downtime start
   ```
-    
+
   2. Add the following parameters to your environment's `public.yml`:
   ```
   elasticsearch_version: 2.4.6
   elasticsearch_download_sha256: 5f7e4bb792917bb7ffc2a5f612dfec87416d54563f795d6a70637befef4cfc6f.
-  ELASTICSEARCH_MAJOR_VERSION: 2
+  ...
+  localsettings:
+    ...
+    ELASTICSEARCH_MAJOR_VERSION: 2
   ```
-    
+
   3. Update the local settings
   ```bash
-  commcare-cloud <env> update-config
+  commcare-cloud ${ENV} update-config
   ```
-  
-  4. Disable the cluster routing
+
+  4. Disable Elasticsearch shard allocation
   ```bash
-  curl -X PUT "<ES-IP>:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
+  curl -X PUT "$(cchq ${ENV} lookup elasticsearch:0):9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
   {
     "persistent": {
-       "cluster.routing.allocation.enable": "none"
+      "cluster.routing.allocation.enable": "none"
     }
   }
   '
   ```
 
   5. Perform a synced flush
   ```bash
-  curl -X POST "<ES-IP>:9200/_flush/synced?pretty"
+  curl -X POST "$(cchq ${ENV} lookup elasticsearch:0):9200/_flush/synced?pretty"  # safe to reissue a few times if it fails
   ```
-    
+
   6. Stop the Monit and Elasticsearch services
   ```bash
-  commcare-cloud <env> run-shell-command all "service monit stop" -b --limit=elasticsearch
-  commcare-cloud <env> run-shell-command all "service elasticsearch stop" -b --limit=elasticsearch
+  commcare-cloud ${ENV} run-shell-command all "service monit stop" -b --limit=elasticsearch
+  commcare-cloud ${ENV} run-shell-command all "service elasticsearch stop" -b --limit=elasticsearch
   ```
-    
+
   7. Install and run the new version of Elasticsearch
   ```bash
-  commcare-cloud <env> ansible-playbook deploy_stack.yml --limit=elasticsearch --tags=elasticsearch
+  commcare-cloud ${ENV} ansible-playbook deploy_stack.yml --limit=elasticsearch --tags=elasticsearch
   ```
-    
+
   8. Stop the Monit and Elasticsearch services
   ```bash
-  commcare-cloud <env> run-shell-command all "service monit stop" -b --limit=elasticsearch
-  commcare-cloud <env> run-shell-command all "service elasticsearch stop" -b --limit=elasticsearch
+  commcare-cloud ${ENV} run-shell-command all "service monit stop" -b --limit=elasticsearch
+  commcare-cloud ${ENV} run-shell-command all "service elasticsearch stop" -b --limit=elasticsearch
   ```
-    
-  9. Rename the newly created 2.4.6 data directory with diffrent name and then rename the 1.7.6 data folder with new version 2.4.6
+
+  9. Rename the newly created 2.4.6 data directory so it's out of the way, and make a backup of the 1.7.6 data directory, and then move it to the new 2.4.6 data directory location
```

Comment:
If we're recommending a backup, shouldn't it be done earlier?  Like right after bringing down the site?

Reply:
Also good idea, ✔️